Presentation.Web/Controllers/API/V1/OrganizationController.cs

```diff
@@ -37,39 +42,24 @@ public virtual HttpResponseMessage Get([FromUri] string q, [FromUri] PagingModel
             return GetAll(paging);
         }
 
-        public HttpResponseMessage GetBySearch(string q, int orgId, int take = 25)
+        public HttpResponseMessage GetBySearchAndOrganizationId(string q, bool onlyWhereUserHasMembership, [FromUri] V1BoundedPaginationQuery pagination = null)
         {
-            try
-            {
-                q = HttpUtility.UrlDecode(q);
-                var canSeeAll = GetCrossOrganizationReadAccessLevel() == CrossOrganizationDataReadAccessLevel.All;
-                var userId = UserId;
-                var dtos =
-                    Repository
-                        .AsQueryable()
-                        .Where(org => org.Name.Contains(q) || org.Cvr.Contains(q))
-                        .Where(org => canSeeAll || org.ObjectOwnerId == userId ||
-                                      // it's public everyone can see it
-                                      org.AccessModifier == AccessModifier.Public ||
-                                      // everyone in the same organization can see normal objects
-                                      org.AccessModifier == AccessModifier.Local &&
-                                      org.Id == orgId ||
-                                      org.OrgUnits.Any(x => x.Rights.Any(y => y.UserId == userId)))
-                        .AsEnumerable()
-                        .Where(AllowRead)
-                        .OrderBy(_ => _.Name)
-                        .Take(take)
-                        .MapToShallowOrganizationDTOs()
-                        .ToList();
-
-                return Ok(dtos);
-            }
-            catch (Exception e)
-            {
-                return LogError(e);
-            }
-        }
+            q = HttpUtility.UrlDecode(q);
+
+            var refinements = new List<IDomainQuery<Organization>>();
 
+            if (!string.IsNullOrWhiteSpace(q))
+                refinements.Add(new QueryByNameOrCvrContent(q));
+            
+            return _organizationService
+                .SearchAccessibleOrganizations(onlyWhereUserHasMembership, refinements.ToArray())
```

Comment:
first `onlyWhereUserHasMembership?` param should be hard coded to false in this call